src/main/java/erb/RouteImpl.java

```diff
@@ -15,61 +15,111 @@
  */
 public class RouteImpl implements Route {
 	// erb: vk US-57-TASK-106 changed node to routeName
-	private String routeId;
+	private int routeId;
 	private String routeName;
-	private String length;
-	private String duration;
+	//erb: vk US-57-TASK-117  call Devin's code to get the actual path
+	// and set length and duration
+	private double length;
+	private double duration;
+    private String path;
+
 
 	// erb: vk US-55-TASK-112 adding sortTag to be able to sort by each column
 	private int sortTag = _SORTBY_ROUTE_ID;
 
 	private static Integer routeCount = 1;
 
 	/*
-	 * Contructor when all fields are passed in so the routeId can be incremented
+	 * Constructor when all fields are passed in so the routeId can be incremented
 	 */
-	public RouteImpl(String name, String length, String duration, String routeId) {
-		this.routeId = routeId;
-		this.routeName = name;
-		this.length = length;
-		this.duration = duration;
+	public RouteImpl(int i, String n, String p, double d, double l) {
+		this.routeId = i;
+		this.routeName = n;
+		this.length = l;
+		this.duration = d;
+		
+		// Placeholder for US-57-110
+		// must call the proper functions to implement the length, duration
 
 		// checks routeId value and increments if needed
-		if (routeCount <= Integer.parseInt(routeId)) {
-			routeCount = Integer.parseInt(routeId) + 1;
+		if (routeCount <= routeId) {
+			routeCount = routeId + 1;
 		}
-
 	}
-	// erb: US-55-TASK-112 from mc refactor to standarized XImpl constructors
-	public RouteImpl(String routeName, String length, String duration) {
-		this(routeName, length, duration, routeCount.toString());
+
+	// erb: US-57-TASK-117 - only setting routeNname and path (string of node ids)
+	// erb: VK never called
+	public RouteImpl(String routeName, String path, double length, double duration) {
+	    this.routeName = routeName;
+        this.path = path;
+        this.length = length;
+        this.duration = duration;
+        this.path = path;
+        routeCount.toString();
+        
+        routeId++;
 	}
+	
 
-	public String getRouteId() {
+    // erb: US-57-TASK-117 used by UI
+	public RouteImpl(String routeName, String path) {
+        this.routeName = routeName;
+        this.path = path;
+        System.out.println("Path: " + path);
+        routeCount.toString();
+        
+        routeId = routeCount++;
+        // Placeholder for US-57-110
+        // must call the proper functions to implement the 'true' path the length, duration
+        createPath(path);
+        
+        routeCount.toString();
+    }
+
+	private void createPath(String p) {
+        // TODO erb: vk placeholder for US-57-TASK-110
+	    // fake method to calculate actual path with length and duration
+	    this.setPath("2,3,4");
```

Comment:
This should be a real path so in demo creating a new route will show the true future functionality
```suggestion
	    this.setPath("1,2,60,6,5,4,3,1");
```

Reply:
Ok :)